crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -3188,11 +3188,35 @@ mod tests {
         assert!(found);
     }
 
+    // Test that a single query can't both contain a mutable reference to a
+    // component C and an `EntityRefExcept` that doesn't include C among its
+    // exclusions.
+    #[test]
+    #[should_panic]
+    fn entity_ref_except_conflicts_with_self() {
+        let mut world = World::new();
+        world.init_component::<TestComponent>();
+        world.init_component::<TestComponent2>();
+
+        world.spawn(TestComponent(0)).insert(TestComponent2(0));
+
+        // This should panic, because we have a mutable borrow on
+        // `TestComponent` but have a simultaneous indirect immutable borrow on
+        // that component via `EntityRefExcept`.
+        let system = (QueryParamBuilder::new(|builder| {
+            builder.data::<(&mut TestComponent, EntityRefExcept<TestComponent2>)>();
+        }),)
+            .build_state(&mut world)
+            .build_system(|_: Query<(&mut TestComponent, EntityRefExcept<TestComponent2>)>| {});
```

Comment:
FYI this is no different than a plain system taking a `Query<(&mut TestComponent, EntityRefExcept<TestComponent2>)>`. `builder.data` is doing nothing here because the data for the static type in the query is pre-registered.

Reply:
Yeah, world.run_system_once is probably going to be easier.